pint_models/database.py

```diff
@@ -0,0 +1,210 @@
+# Copyright (c) 2024 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of version 3 of the GNU General Public License as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.   See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE about this file by physical or electronic mail,
+# you may find current contact information at www.suse.com
+
+# NOTE(gyee):
+# see https://flask.palletsprojects.com/en/1.1.x/patterns/sqlalchemy/
+
+import logging
+import os
+import re
+
+from sqlalchemy import create_engine
+from sqlalchemy.orm import scoped_session, sessionmaker
+from pint_models.models import Base
+
+
+def get_environ_or_bust(key_name):
+    assert key_name in os.environ, 'Environment variable %s is required.' % (
+        key_name)
+    return os.environ.get(key_name)
+
+
+def create_db_logger(outputfile):
+    """Function to setup logging of SQL statements
+
+    Args:
+        outputfile (filepath): The filepath where the SQL statements will be
+            logged
+    """
+    if not outputfile:
+        return
+
+    db_log_file_name = outputfile
+    db_handler_log_level = logging.INFO
+    db_logger_log_level = logging.DEBUG
+
+    db_handler = logging.FileHandler(db_log_file_name)
+    db_handler.setLevel(db_handler_log_level)
+
+    db_logger = logging.getLogger('sqlalchemy')
+    db_logger.addHandler(db_handler)
+    db_logger.setLevel(db_logger_log_level)
+
+
+def _create_postgres_url(db_user, db_password, db_name, db_host,
```

Comment:
It be nice to move the internal functions to the end rather than having them intermixed with the "public" functions.